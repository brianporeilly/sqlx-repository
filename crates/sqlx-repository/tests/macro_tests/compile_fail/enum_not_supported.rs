//! Test that repository derive fails on non-struct types

use sqlx_repository::prelude::*;

#[derive(Debug, Clone, Repository)]
#[repository(table = "users")]
pub enum User {  // This should fail - Repository only works on structs
    Active,
    Inactive,
}

fn main() {}