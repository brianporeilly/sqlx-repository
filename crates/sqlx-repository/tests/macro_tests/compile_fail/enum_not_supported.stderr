error: Repository derive only supports structs, not enums.

       Example:
       #[derive(Repository)]
       #[repository(table = "users")]
       pub struct User {
           pub id: i32,
           pub name: String,
       }
  --> tests/macro_tests/compile_fail/enum_not_supported.rs:6:1
   |
6  | / #[repository(table = "users")]
7  | | pub enum User {  // This should fail - Repository only works on structs
8  | |     Active,
9  | |     Inactive,
10 | | }
   | |_^
